src/id-map.ts

```diff
@@ -0,0 +1,51 @@
+import { JSORMBase } from "./model"
+import { Association } from "./associations"
+import { cloneDeep } from "./util/clonedeep"
+
+export class IDMap {
```

Comment:
Can't think of a reason this would be an improvement but had the thought so wanted to mention it: any reason this should be model-specific, e.g. `new IDMap(Post)` instead of shared from a root `ApplicationRecord`?

Reply:
Only reason I can think of would be performance, since it's a smaller object lookup. Maybe keep as is unless that becomes an issue.